docs/HyperIndex/Guides/rpc-requests.md

```diff
@@ -0,0 +1,324 @@
+
+We then make some light modifications to remove unnecessary events and simplify the schema. The resulting config, schema, and event handlers look as follows:
+
+> config.yaml
+
+:::info
+We remove the `FeeAmountEnabled` and `OwnerChanged` events as they are not relevant to our use case.
+:::
+
+```yaml
+name: uniswap-v3-factory-token-indexer
+networks:
+- id: 1
+  start_block: 12369620
+  contracts:
+  - name: UniswapV3Factory
+    address:
+    - 0x1F98431c8aD98523631AE4a59f267346ea31F984
+    handler: src/EventHandlers.ts
+    events:
+    - event: PoolCreated(address indexed token0, address indexed token1, uint24 indexed fee, int24 tickSpacing, address pool)
+rollback_on_reorg: false
+```
+
+> shema.graphql
+
+:::info
+Simplify our schema to only include the pool and token information.
+:::
+
+```graphql
+type Token {
+  id: ID! #address
+  name: String!
+  symbol: String!
+  decimals: Int!
+}
+
+type Pool {
+  id: ID! # address
+  token0: Token! 
+  token1: Token!
+  fee: BigInt!
+  tickSpacing: BigInt!
+  pool: String!
+}
+```
+
+> src/EventHandlers.ts
+
+```typescript
+import {
+  UniswapV3Factory,
+  Pool,
+} from "generated";
+import { getTokenDetails } from "./tokenDetails";
+
+UniswapV3Factory.PoolCreated.handler(async ({ event, context }) => {
+  const entity: Pool = {
+    id: `${event.chainId}_${event.block.number}_${event.logIndex}`,
+    token0_id: event.params.token0,
+    token1_id: event.params.token1,
+    fee: event.params.fee,
+    tickSpacing: event.params.tickSpacing,
+    pool: event.params.pool,
+  };
+  context.Pool.set(entity)
+
+
+  try {
+    const {name: name0, symbol: symbol0, decimals: decimals0} = await getTokenDetails(event.params.token0, event.chainId);
+    context.Token.set({
+      id: event.params.token0,
+      name: name0,
+      symbol: symbol0,
+      decimals: decimals0
+    })
+  } catch (error) {
+    console.log(error)
+    return
+  }
+
+  try {
+    const {name: name1, symbol: symbol1, decimals: decimals1} = await getTokenDetails(event.params.token0, event.chainId);
+    context.Token.set({
+      id: event.params.token1,
+      name: name1,
+      symbol: symbol1,
+      decimals: decimals1,
+    })
+  } catch (error) {
+    console.log(error)
+    return
+  }
+});
+
+```
+
+### Part 2: Fetch token information
+
+Note how we use the `multicall` feature from Viem to batch our 3 queries into one request to improve performance and help avoid rate limiting. Very useful!
+
+Here, we use an alchemy API key to make RPC requests to the ethereum mainnet. You can get an alchemy API key [here](https://www.alchemy.com/) and add it to your `.env` file. Alternatively, use any other rpc provider of your choice.
+
+> src/tokenDetails.ts
+
+```typescript
+import { createPublicClient, getContract, http } from 'viem'
+import { mainnet } from 'viem/chains'
+import { Cache, CacheCategory } from "./cache";
+import { ERC20ABI } from './constants';
+
+const apiKey = process.env.ALCHEMY_API_KEY;
```

Comment:
I suggest rather making this the full rpc endpoint than an api key as it's more flexible to switch between providers. ie something like
```suggestion
const apiKey = process.env.RPC_URL;
```